File: app/src/main/java/com/lelloman/demo/fragment/classicidenticon/ClassicIdenticonFragment.java
package com.lelloman.demo.fragment.classicidenticon;

import com.lelloman.demo.fragment.IdenticonFragment;
import com.lelloman.identicon.drawable.IdenticonDrawable;
import com.lelloman.identicon.drawable.ClassicIdenticonDrawable;


public abstract class ClassicIdenticonFragment extends IdenticonFragment {

	@Override
	protected IdenticonFragment.RecyclerViewAdapter getAdapter() {
		return new IdenticonFragment.RecyclerViewAdapter() {
			@Override
			protected int getHashForPosition(int position) {
				return ClassicIdenticonFragment.this.getHashForPosition(position);
			}

			@Override
			protected IdenticonDrawable makeIdenticonDrawable(int width, int height, int hash) {
				return new ClassicIdenticonDrawable(width, height, hash);
			}

		};
	}

	protected abstract int getHashForPosition(int position);
}

File: app/src/main/java/com/lelloman/demo/fragment/classicidenticon/ClassicIdenticonFragment.kt
package com.lelloman.demo.fragment.classicidenticon

import com.lelloman.demo.fragment.IdenticonFragment
import com.lelloman.identicon.drawable.ClassicIdenticonDrawable

abstract class ClassicIdenticonFragment : IdenticonFragment() {

    override open val adapter = object : IdenticonFragment.RecyclerViewAdapter() {
        override fun getHashForPosition(position: Int) =
            this@ClassicIdenticonFragment.getHashForPosition(position)

        override fun makeIdenticonDrawable(width: Int, height: Int, hash: ByteArray) =
            ClassicIdenticonDrawable(width, height, hash)
    }

    protected abstract fun getHashForPosition(position: Int): Int
}
